services/src/test-library/src/main/java/org/openkilda/testing/service/database/DatabaseNeoImpl.java

```diff
@@ -0,0 +1,91 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.testing.service.database;
+
+import org.openkilda.testing.model.topology.TopologyDefinition.Isl;
+
+import com.google.common.collect.ImmutableMap;
+import org.neo4j.driver.v1.Driver;
+import org.neo4j.driver.v1.GraphDatabase;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class DatabaseNeoImpl implements DisposableBean, Database {
+    private Driver neo;
+
+    public DatabaseNeoImpl(@Value("${neo.uri}") String neoUri) {
+        neo = GraphDatabase.driver(neoUri);
+    }
+
+    @Override
+    public void destroy() {
+        if (neo != null) {
+            neo.close();
+        }
+    }
+
+    /**
+     * Updates certain property on a certain ISL.
+     *
+     * @return true if at least 1 ISL was affected
+     */
+    @Override
+    public boolean updateLinkProperty(Isl isl, String property, Object value) {
+        String query = "MATCH ()-[link:isl {src_port:$srcPort, dst_port:$dstPort, src_switch:$srcSwitch, "
+                + "dst_switch:$dstSwitch}]->() SET link += {props}";
+        Map<String, Object> params = getParams(isl);
+        params.put("props", ImmutableMap.of(property, value));
+        StatementResult result;
+        try (Session session = neo.session()) {
+            result = session.run(query, params);
+        }
+        return result.summary().counters().propertiesSet() > 0;
+    }
+
+    /**
+     * Set ISL's max bandwidth to be equal to its speed (the default situation).
+     *
+     * @param isl ISL to be changed
+     * @return true if at least 1 ISL was affected
+     */
+    @Override
+    public boolean revertIslBandwidth(Isl isl) {
+        String query = "MATCH ()-[link:isl {src_port:$srcPort, dst_port:$dstPort, src_switch:$srcSwitch, "
+                + "dst_switch:$dstSwitch}]->() SET link.max_bandwidth=link.speed, link.available_bandwidth=link.speed";
+        Map<String, Object> params = getParams(isl);
+        StatementResult result;
+        try (Session session = neo.session()) {
+            result = session.run(query, params);
+        }
+        return result.summary().counters().propertiesSet() > 0;
+    }
+
+    private Map<String, Object> getParams(Isl isl) {
+        Map<String, Object> params = new HashMap<>();
```

Comment:
We could add the initial size for the map.

Reply:
done